Sources/StreamChatUI/ChatMessageList/ChatMessage/GiftPacketSentBubble.swift

```diff
@@ -0,0 +1,356 @@
+        detailsStack.transform = .mirrorY
+        detailsStack.alignment = .leading
+        detailsStack.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            detailsStack.leadingAnchor.constraint(equalTo: subContainer.leadingAnchor, constant: 10),
+            detailsStack.trailingAnchor.constraint(equalTo: subContainer.trailingAnchor, constant: -31),
+            detailsStack.bottomAnchor.constraint(equalTo: stickerContainer.topAnchor, constant: -8),
+        ])
+        detailLabel.translatesAutoresizingMaskIntoConstraints = false
+        subContainer.addSubview(detailLabel)
+        NSLayoutConstraint.activate([
+            detailLabel.leadingAnchor.constraint(equalTo: subContainer.leadingAnchor, constant: 10),
+            detailLabel.trailingAnchor.constraint(equalTo: subContainer.trailingAnchor, constant: 0),
+            detailLabel.bottomAnchor.constraint(equalTo: detailsStack.topAnchor, constant: 0),
+            detailLabel.topAnchor.constraint(equalTo: subContainer.topAnchor, constant: 10)
+        ])
+        detailLabel.transform = .mirrorY
+        detailLabel.textAlignment = .left
+
+        timestampLabel.translatesAutoresizingMaskIntoConstraints = false
+        viewContainer.addSubview(timestampLabel)
+
+        NSLayoutConstraint.activate([
+            timestampLabel.leadingAnchor.constraint(equalTo: viewContainer.leadingAnchor, constant: 0),
+            timestampLabel.trailingAnchor.constraint(equalTo: viewContainer.trailingAnchor, constant: 0),
+            timestampLabel.bottomAnchor.constraint(equalTo: subContainer.topAnchor, constant: -8),
+            timestampLabel.topAnchor.constraint(equalTo: viewContainer.topAnchor, constant: 0),
+            timestampLabel.heightAnchor.constraint(equalToConstant: 15)
+        ])
+        timestampLabel.transform = .mirrorY
+    }
+
+    func configureCell(isSender: Bool) {
+        self.isSender = isSender
+        setBubbleConstraints(isSender)
+        setFlair()
+    }
+
+    private func setBubbleConstraints(_ isSender: Bool) {
+        leadingAnchorForSender?.isActive = isSender
+        leadingAnchorForSender?.constant = cellWidth
+        trailingAnchorForSender?.isActive = isSender
+        trailingAnchorForSender?.constant = -8
+        leadingAnchorForReceiver?.isActive = !isSender
+        leadingAnchorForReceiver?.constant = 8
+        trailingAnchorForReceiver?.isActive = !isSender
+        trailingAnchorForReceiver?.constant = -cellWidth
+        timestampLabel.textAlignment = isSender ? .right : .left
+    }
+
+    private var cellWidth: CGFloat {
+        return UIScreen.main.bounds.width * 0.3
+    }
+
+    private func setFlair() {
+        guard let extraData = content?.extraData,
+        let flair = extraData.flair else {
+            return
+        }
+        stickerContainer.removeAllArrangedSubviews()
+        if self.isVideoType(flair) {
+            NotificationCenter.default.removeObserver(self,
```

Comment:
@tunguyen-zien  it would be good to have `NotificationCenter.default.addObserver` code on init method.

Reply:
hi @parth-simformsolutions because we just use NotificationCenter.default.addObserver in case Video (besides in case gif ) so I think we don't need to add on init method.